File: backend/app/schemas/docusign.py
"""DocuSign eSignature request/response schemas."""

from uuid import UUID

from pydantic import BaseModel

from app.schemas.base import Str50, Str255, Str2000


class CreateEnvelopeRequest(BaseModel):
    document_id: UUID
    signer_email: Str255
    signer_name: Str255
    return_url: Str2000


class EnvelopeResponse(BaseModel):
    envelope_id: Str255
    document_id: UUID
    docusign_status: Str50


class SigningUrlResponse(BaseModel):
    signing_url: Str2000
    envelope_id: Str255

File: frontend/src/lib/api/docusign.ts
import api from "@/lib/api";

export interface CreateEnvelopeRequest {
  document_id: string;
  signer_email: string;
  signer_name: string;
  return_url: string;
}

export interface EnvelopeResponse {
  envelope_id: string;
  document_id: string;
  docusign_status: string;
}

export interface SigningUrlResponse {
  signing_url: string;
  envelope_id: string;
}

export async function createDocuSignEnvelope(
  data: CreateEnvelopeRequest,
): Promise<EnvelopeResponse> {
  const res = await api.post<EnvelopeResponse>("/api/v1/docusign/envelopes", data);
  return res.data;
}

export async function getDocuSignSigningUrl(params: {
  document_id: string;
  signer_email: string;
  signer_name: string;
  return_url: string;
}): Promise<SigningUrlResponse> {
  const res = await api.get<SigningUrlResponse>("/api/v1/docusign/signing-url", {
    params,
  });
  return res.data;
}
